### ISSUE TITLE
Inference Speed

### ISSUE THREAD

**Issue by jdc711** [openvinotoolkit/openvino#9784]
Hi, I had a question about execution time on Neural Compute Stick 2:

I was hoping to get some inference execution speed boost by running a simple neural network model on the NC2 stick instead of CPU. However, my initial experiment showed that the performance on myriad neural compute stick 2 is MUCH slower than CPU:

For MYRIAD NC2 stick:
[ INFO ] First inference took 33.88 ms
[Step 11/11] Dumping statistics report
Count: 2596 iterations
Duration: 60141.63 ms
Latency: 92.60 ms
Throughput: 5525.09 FPS

For CPU:
[ INFO ] First inference took 17.07 ms
[Step 11/11] Dumping statistics report
Count: 148124 iterations
Duration: 60001.79 ms
Latency: 1.61 ms
Throughput: 315988.43 FPS

My model is a simple linear model with input format: [128, 1, 28, 28], which gets flattened in forward to -> [128,784]:
def init(self):
super(NeuralNetwork, self).init()
self.layers = nn.Sequential(
nn.Linear(28*28, 512),
nn.ReLU(),
nn.Linear(512, 200),
nn.ReLU(),
nn.Linear(200, 10),
)

def forward(self, x):
    # Flatten x: [128, 1, 28, 28] -> [128, 784]
    x = x.view(x.shape[0], -1)
    output = self.layers(x)
    return output

I am running benchmark_app.py with -m model.xml, with random input generated (no bin file provided in command)

Would you possibly have any ideas as to why my performance on NC2 is so much slower than the CPU?

For the CPU used, I am using 2.6 GHz 6-Core Intel Core i7, for Macbook pro. This is the current execution time, throughput, etc that is attained using CPU vs MYRIAD NCS2 Stick:

**Comment by Iffa-Meah** [comment]
The performance on CPU is expected to be better compared to NCS2 since CPU has more powerful computing power.

NCS2 is just an accelerator device that would help in a certain situation, especially when additional computing power is required.

You may refer here for more comparison: [OpenVINO Performance Benchmark Results](https://docs.openvino.ai/latest/openvino_docs_performance_benchmarks_openvino.html#doxid-openvino-docs-performance-benchmarks-openvino)

Besides, CPU required FP32 model format while NCS2 required FP16 model format. FP16 **might** have something called Quantization Error since it is squeezed from a full precision model which made it smaller in size. This would somewhat affect accuracy/performance.

You may refer here for further understanding: [Choosing the right precision](https://www.youtube.com/watch?v=RF8ypHyiKrY)

Generally, performance means how fast the model is in deployment with two key metrics to measure performance: **latency** and **throughput**.

In OpenVINO, there are 2 ways to enhance the performance:
1. During development: Post-training Optimization tool (POT), Neural Network Compression Framework (NNCF), Model Optimizer
2. During deployment: tuning inference parameters and optimizing model execution

It's also possible to combine both ways.
You may refer here: [Performance Optimization Guide](https://docs.openvino.ai/latest/openvino_docs_optimization_guide_dldt_optimization_guide.html?)

**Comment by Iffa-Meah** [comment]
Closing issue, feel free to re-open or start a new issue if additional assistance is needed.
